### PR TITLE
Modify CI/CD for static packages

### PR DESCRIPTION
**Pull request by foxzi**
Modify .pack.mk for deploying static packages. Change the CI/CD process for creating and publishing packages to the repository. 

This PR connected with https://github.com/tarantool/rws/pull/79. It is needed to support static packages by RWS.

### CODE REVIEW

**Comment by Totktonada** [comment]
I can't actually review the infrastructural details. I have no enough context. (BTW, this change looks coupled with https://github.com/tarantool/rws/pull/79, but it is not mentioned anywhere.)

If my review is needed for the design decisions (such as public URIs of the new repositories), I would like to read a summary of the changes (and what is not to be changed -- source tarballs location, modules repository?) in some form (say, an RFC document). For example, the previous change was made after a thorough discussion around [this](https://github.com/orgs/tarantool/discussions/6182#user-content-delivery-channels) section of the new release policy document.

File: .github/workflows/static_build_packaging.yml [hunk 1]
```diff
@@ -62,8 +62,11 @@ jobs:
       - uses: ./.github/actions/environment
       - name: Install deps
         uses: ./.github/actions/install-deps-debian
-      - name: Build static packages
-        run: make -f .pack.mk package-static
+      - name: packaging
+        env:
+          RWS_AUTH: ${{ secrets.RWS_AUTH }}
+          OUTPUT_DIR: ../build
```
Comment (ylobankov) [inline .github/workflows/static_build_packaging.yml:68]: I think we can omit this.

File: .github/actions/pack-and-deploy/action.yml [hunk 2]
```diff
@@ -9,8 +9,13 @@ runs:
         # on it: app-tap/gh-4983-tnt-e-assert-false-hangs.test.lua.
         export PACKPACK_EXTRA_DOCKER_RUN_PARAMS='--init'
 
-        # Create packages.
-        make -f .pack.mk package
+        # Create packages. For versions 3.x creates static packages.
+        # For old versions 1.x and 2.x creates packages of specific OS.
+        if ${{ (startsWith(github.ref, 'refs/tags/3.') }}; then
```
Comment (ylobankov) [inline .github/actions/pack-and-deploy/action.yml:14]: There is a typo. Please remove `(` before `startsWith`.

File: .pack.mk [hunk 3]
```diff
@@ -7,6 +7,7 @@ RWS_BASE_URL = https://rws.tarantool.org
 PRODUCT_NAME = tarantool
 
 VARDIR ?=/tmp/t
+OUTPUT_DIR ?= ../build/
```
[the remark this answers is not in the document]
Comment (foxzi) [reply .pack.mk]: fixed

File: .pack.mk [hunk 4]
```diff
@@ -54,35 +55,33 @@ package-static:
 		export VERSION="$$(echo ${GIT_DESCRIBE} | sed ${SED_REPLACE_VERSION_REGEX} | sed 's/-/~/').dev"; \
 	fi; \
 	echo VERSION=$${VERSION}; \
+	export OUTPUT_DIR=${OUTPUT_DIR}; \
```
[the remark this answers is not in the document]
Comment (foxzi) [reply .pack.mk:57]: done

File: .github/workflows/static_build_packaging.yml (hunk 1 above)
Comment (foxzi) [reply .github/workflows/static_build_packaging.yml]: removed

File: .github/actions/pack-and-deploy/action.yml (hunk 2 above)
Comment (foxzi) [reply .github/actions/pack-and-deploy/action.yml]: outdated

File: .github/actions/pack-and-deploy/action.yml [hunk 5]
```diff
@@ -9,8 +9,13 @@ runs:
         # on it: app-tap/gh-4983-tnt-e-assert-false-hangs.test.lua.
         export PACKPACK_EXTRA_DOCKER_RUN_PARAMS='--init'
 
-        # Create packages.
-        make -f .pack.mk package
+        # Create packages. For versions 3.x creates static packages.
```
[the remark this answers is not in the document]
Comment (foxzi) [reply .github/actions/pack-and-deploy/action.yml:12]: done

**Review by ylobankov — APPROVED** [review]

**Pull request merged**